pkg/admission/namespaceconditions/decorator.go

```diff
@@ -0,0 +1,45 @@
+package namespaceconditions
+
+import (
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apiserver/pkg/admission"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+	corev1lister "k8s.io/client-go/listers/core/v1"
+)
+
+// NamespaceLabelConditions provides a decorator that can delegate and conditionally add label conditions
+type NamespaceLabelConditions struct {
+	Delegate admission.Decorator
```

Comment:
concatenation of decorators looks orthogonal. Why this delegate?

I would expect `admission.Concat(name NamespaceLabelConditions.WithNamespaceLabelConditions, someOtherDecorator, yetAnotherDecorator)` if you really want to avoid passing `name` multiple time in a manual concatenation.

Reply:
> concatenation of decorators looks orthogonal. Why this delegate?
> 
> I would expect admission.Concat(name NamespaceLabelConditions.WithNamespaceLabelConditions, someOtherDecorator, yetAnotherDecorator) if you really want to avoid passing name multiple time in a manual concatenation.

Yeah, I can make a decorator chain upstream for us to get in 1.10.  Functionally acceptable otherwise?

Reply:
Yes, am a big fan of decorators. And we obviously need this.